libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -346,10 +346,8 @@ bool AP_Mission::is_nav_cmd(const Mission_Command& cmd)
 ///     accounts for do_jump commands but never increments the jump's num_times_run (advance_current_nav_cmd is responsible for this)
 bool AP_Mission::get_next_nav_cmd(uint16_t start_index, Mission_Command& cmd)
 {
-    uint16_t cmd_index = start_index;
-
     // search until the end of the mission command list
-    while(cmd_index < (unsigned)_cmd_total) {
+    for (uint16_t cmd_index = start_index; cmd_index < (unsigned)_cmd_total; cmd_index++) {
```

Comment:
Interesting. This is preexisting, but it has an actual bug. Consider the following mission that ends with the following items:

1. DO_CAM_TRIGG_DIST
1. NAV_WAYPOINT
1. NAV_WAYPOINT
1. DO_JUMP to 1.

Now start looking from index 4 (the jump) the next command will select item 1, which is fine, it's not a nav_cmd, but we've now looped past cmd_total and ignore that the vehicle should go to waypoint 2. (My numbering might be off here, but I believe the issue is valid (albeit rare))